make xnode fails with AttributeError: 'PBXAggregateTarget' object has no attribute 'FrameworksPhase'

I'm using node v5.0.0.

```
~/Development/mapbox-gl-native $ make xnode
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.6.14
node-pre-gyp info using node@5.0.0 | darwin | x64
gyp info it worked if it ends with ok
gyp info using node-gyp@0.12.1
gyp info using node@5.0.0 | darwin | x64
(node) child_process: options.customFds option is deprecated. Use options.stdio instead.
gyp info spawn python
gyp info spawn args [ '/usr/local/lib/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-Dhost=osx',
gyp info spawn args   '-Iconfig/osx-x86_64.gypi',
gyp info spawn args   '-Dplatform_lib=osx',
gyp info spawn args   '-Dhttp_lib=none',
gyp info spawn args   '-Dasset_lib=none',
gyp info spawn args   '-Dcache_lib=none',
gyp info spawn args   '-Dheadless_lib=cgl',
gyp info spawn args   '-Dtest=1',
gyp info spawn args   '-Drender=1',
gyp info spawn args   '-Dcxx_host=',
gyp info spawn args   '--depth=.',
gyp info spawn args   '-Goutput_dir=.',
gyp info spawn args   '--generator-output=./build/osx-x86_64',
gyp info spawn args   '-f',
gyp info spawn args   'xcode',
gyp info spawn args   '-Dlibuv_cflags=',
gyp info spawn args   '-Dlibuv_ldflags=',
gyp info spawn args   '-Dlibuv_static_libs=',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/john/Development/mapbox-gl-native/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/john/.node-gyp/5.0.0/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/john/.node-gyp/5.0.0',
gyp info spawn args   '-Dmodule_root_dir=/Users/john/Development/mapbox-gl-native',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
Traceback (most recent call last):
  File "/usr/local/lib/node_modules/node-gyp/gyp/gyp_main.py", line 18, in <module>
    sys.exit(gyp.script_main())
  File "/usr/local/lib/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 534, in script_main
    return main(sys.argv[1:])
  File "/usr/local/lib/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 527, in main
    return gyp_main(args)
  File "/usr/local/lib/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 512, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "/usr/local/lib/node_modules/node-gyp/gyp/pylib/gyp/generator/xcode.py", line 1177, in GenerateOutput
    xct.FrameworksPhase().AddFile(library)
AttributeError: 'PBXAggregateTarget' object has no attribute 'FrameworksPhase'
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/node-gyp/lib/configure.js:337:16)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.0.0
gyp ERR! command "/usr/local/Cellar/node/5.0.0/bin/node" "/usr/local/bin/node-gyp" "configure" "--clang" "--depth=.=-Goutput_dir=." "--generator-output=./build/osx-x86_64=-f" "--module=/Users/john/Development/mapbox-gl-native/lib/mapbox-gl-native.node" "--module_name=mapbox-gl-native" "--module_path=/Users/john/Development/mapbox-gl-native/lib" "--" "-Dhost=osx" "-Iconfig/osx-x86_64.gypi" "-Dplatform_lib=osx" "-Dhttp_lib=none" "-Dasset_lib=none" "-Dcache_lib=none" "-Dheadless_lib=cgl" "-Dtest=1" "-Drender=1" "-Dcxx_host=" "--depth=." "-Goutput_dir=." "--generator-output=./build/osx-x86_64" "-f" "xcode" "-Dlibuv_cflags=" "-Dlibuv_ldflags=" "-Dlibuv_static_libs="
gyp ERR! cwd /Users/john/Development/mapbox-gl-native
gyp ERR! node -v v5.0.0
gyp ERR! node-gyp -v v0.12.1
gyp ERR! not ok 
node-pre-gyp ERR! configure error 
node-pre-gyp ERR! stack Error: Failed to execute 'node-gyp configure --clang --depth=.=-Goutput_dir=. --generator-output=./build/osx-x86_64=-f --module=/Users/john/Development/mapbox-gl-native/lib/mapbox-gl-native.node --module_name=mapbox-gl-native --module_path=/Users/john/Development/mapbox-gl-native/lib -- -Dhost=osx -Iconfig/osx-x86_64.gypi -Dplatform_lib=osx -Dhttp_lib=none -Dasset_lib=none -Dcache_lib=none -Dheadless_lib=cgl -Dtest=1 -Drender=1 -Dcxx_host= --depth=. -Goutput_dir=. --generator-output=./build/osx-x86_64 -f xcode -Dlibuv_cflags= -Dlibuv_ldflags= -Dlibuv_static_libs=' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/john/Development/mapbox-gl-native/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:87:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:818:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
node-pre-gyp ERR! System Darwin 15.0.0
node-pre-gyp ERR! command "/usr/local/Cellar/node/5.0.0/bin/node" "/Users/john/Development/mapbox-gl-native/node_modules/.bin/node-pre-gyp" "configure" "--clang" "--" "-Dhost=osx" "-Iconfig/osx-x86_64.gypi" "-Dplatform_lib=osx" "-Dhttp_lib=none" "-Dasset_lib=none" "-Dcache_lib=none" "-Dheadless_lib=cgl" "-Dtest=1" "-Drender=1" "-Dcxx_host=" "--depth=." "-Goutput_dir=." "--generator-output=./build/osx-x86_64" "-f" "xcode" "-Dlibuv_cflags=" "-Dlibuv_ldflags=" "-Dlibuv_static_libs="
node-pre-gyp ERR! cwd /Users/john/Development/mapbox-gl-native
node-pre-gyp ERR! node -v v5.0.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.14
node-pre-gyp ERR! not ok 
Failed to execute 'node-gyp configure --clang --depth=.=-Goutput_dir=. --generator-output=./build/osx-x86_64=-f --module=/Users/john/Development/mapbox-gl-native/lib/mapbox-gl-native.node --module_name=mapbox-gl-native --module_path=/Users/john/Development/mapbox-gl-native/lib -- -Dhost=osx -Iconfig/osx-x86_64.gypi -Dplatform_lib=osx -Dhttp_lib=none -Dasset_lib=none -Dcache_lib=none -Dheadless_lib=cgl -Dtest=1 -Drender=1 -Dcxx_host= --depth=. -Goutput_dir=. --generator-output=./build/osx-x86_64 -f xcode -Dlibuv_cflags= -Dlibuv_ldflags= -Dlibuv_static_libs=' (1)
make[1]: *** [node/xproj] Error 1
make: *** [Xcode/node] Error 2
```

@kkaefer @mikemorris

--- a/Makefile
+++ b/Makefile
@@ -86,7 +86,7 @@ node: ; $(RUN) HTTP=none ASSET=none CACHE=none Makefile/node
 
 
 .PHONY: Xcode/node
-Xcode/node: ; $(RUN) HTTP=none ASSET=none CACHE=none node/xproj
+Xcode/node: ; $(RUN) HTTP=none ASSET=none CACHE=none Xcode/node
 
 .PHONY: xnode
 xnode: Xcode/node ; @open ./build/binding.xcodeproj

--- a/package.json
+++ b/package.json
@@ -27,6 +27,7 @@
   "devDependencies": {
     "aws-sdk": "^2.2.9",
     "mapbox-gl-test-suite": "mapbox/mapbox-gl-test-suite#744b63be01e36c75c6e629aec16a53048c0b7dbc",
+    "node-gyp": "^3.0.3",
     "read-package-json": "^2.0.2",
     "request": "^2.65.0",
     "tape": "^4.2.1"

--- a/platform/node/src/node_file_source.hpp
+++ b/platform/node/src/node_file_source.hpp
@@ -5,6 +5,7 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wnested-anon-types"
 #include <nan.h>
 #pragma GCC diagnostic pop
 

--- a/platform/node/src/node_log.hpp
+++ b/platform/node/src/node_log.hpp
@@ -5,6 +5,7 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wnested-anon-types"
 #include <nan.h>
 #pragma GCC diagnostic pop
 

--- a/platform/node/src/node_map.hpp
+++ b/platform/node/src/node_map.hpp
@@ -8,6 +8,7 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wnested-anon-types"
 #include <nan.h>
 #pragma GCC diagnostic pop
 

--- a/platform/node/src/node_mapbox_gl_native.cpp
+++ b/platform/node/src/node_mapbox_gl_native.cpp
@@ -1,6 +1,7 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wnested-anon-types"
 #include <node.h>
 #include <nan.h>
 #pragma GCC diagnostic pop

--- a/platform/node/src/node_request.hpp
+++ b/platform/node/src/node_request.hpp
@@ -3,6 +3,7 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wnested-anon-types"
 #include <nan.h>
 #pragma GCC diagnostic pop
 

